task 5

YakovGrischenko opened the pull request:
All tasks from base part were implemented: **+5**
Lambda importProductsFile endpoint [https://jyzag4006l.execute-api.eu-west-1.amazonaws.com/dev/import](https://jyzag4006l.execute-api.eu-west-1.amazonaws.com/dev/import)
link to FE [https://d2l3rxej7y8xn8.cloudfront.net/admin/products](https://d2l3rxej7y8xn8.cloudfront.net/admin/products)
FE PR [https://github.com/YakovGrischenko/nodejs-aws-fe/pull/4](https://github.com/YakovGrischenko/nodejs-aws-fe/pull/4)

Aditional tasks: **+3**
+1  async/await are used in lambda functions
+1  importProductsFile lambda is covered by unit tests **mocked with "aws-sdk-mock"**
+1  moving file to /parsed folder after parsing

My socre: 8 / 8

YakovGrischenko committed:
init: import-service created by template aws-nodejs-typescript

YakovGrischenko committed:
feat:  importProductsFile implemented + dumb for importFileParser

YakovGrischenko committed:
feat: importFileParser implementation

YakovGrischenko committed:
fix: importFileParser works correct in AWS + prettier

YakovGrischenko committed:
fix: minor fixes

YakovGrischenko committed:
udp: unmocked unit tests

YakovGrischenko committed:
feat: unit tests mocked with "aws-sdk-mock"

KatyaAksenchik commented on import-service/src/importFileParser.ts:
```diff
@@ -0,0 +1,67 @@
+export const importFileParser: S3Handler = async (event: S3Event) => {
+  console.log('lambda : importFileParser, event: ', event)
+
+  const s3 = new AWS.S3({ region: 'eu-west-1' })
```
Maybe wrap all of this in `try catch`?

YakovGrischenko replied:
Done. [3077edc](https://github.com/YakovGrischenko/nodejs-aws-be/pull/3/commits/3077edc16731328be56cf30709bf6943197e75a4)

YakovGrischenko committed:
upd: additional try/catch in importFileParser lambda

YakovGrischenko commented:
![Screenshot 2020-11-18 at 00 11 01](https://user-images.githubusercontent.com/1833936/99425023-edd30580-2934-11eb-9583-7d3a7feec609.png)

Fixed this on FE side. This issue can be reproduced only for Windows machines. Fixed anyway.
